enclave-runtime/src/lib.rs

```diff
@@ -578,6 +582,79 @@ where
 	Ok(())
 }
 
+//For now get the DOT/currency exchange rate from coingecko API
+#[no_mangle]
+pub unsafe extern "C" fn update_market_data_xt(
+	currency: *const u8,
+	currency_size: u32,
+	unchecked_extrinsic: *mut u8,
+	unchecked_extrinsic_size: u32,
+) -> sgx_status_t {
+	let extrinsic_slice =
+		slice::from_raw_parts_mut(unchecked_extrinsic, unchecked_extrinsic_size as usize);
+
+	let mut curr_slice = slice::from_raw_parts(currency, currency_size as usize);
+	let curr: String = Decode::decode(&mut curr_slice).unwrap();
+	let xts = match update_market_data_internal::<Block>(curr) {
+		Ok(xts) => xts,
+		Err(_) => return sgx_status_t::SGX_ERROR_UNEXPECTED,
+	};
+
+	//only one extrinsic to send over the node api directly. No need of ocall
```

Comment:
Hm. I know what you mean with that comment, but if you're reading this without any previous discussion background, that comment will confuse I think

Maybe just add a comment above function `write_slice_and_whitespace_pad` like
`// Save created extrinsic as slice in the return value unchecked_extrinsic.`